src/main/java/com/aws/greengrass/secretmanager/SecretManager.java

```
@@ -329,23 +329,30 @@ private GetSecretValueResponse getSecret(String secretId, String versionId, Stri
      * @throws GetSecretException when secret is not found
      */
     public String validateSecretId(String secretId) throws GetSecretException {
-        String arn = secretId;
         if (Utils.isEmpty(secretId)) {
             throw new GetSecretException(400, "SecretId absent in the request");
         }
-        // normalize name to arn
-        if (!Pattern.matches(VALID_SECRET_ARN_PATTERN, secretId)) {
-            arn = nameToArnMap.get(secretId);
-            if (arn == null) {
+        try {
+            return getArnFromCache(secretId);
+        } catch (GetSecretException e) {
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While re-throwing the caught exception with modifications , information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 341.

---

nit: we should log this exception message out as a debug somewhere